operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -1753,6 +1792,82 @@ def isSSLListenPortEnabled(ssl, domain):
       enabled = True
   return enabled
 
+def getSSLPortIfEnabled(server, domain, is_server_template=True):
+  """
+  return the SSL listen port if enabled -
+    If SSL is enabled:
+      If is_server_template is False then just return the SSL listen port from server mbean.
+      If is_server_template is True then return the actual SSL listen port that it listens on.
+
+    If SSL is not configured but domain has SecureMode enabled return 7002.
+    If SSL is configured but explicitly disabled, return None.
+
+  :param server: server or server template
+  :param domain: domain mbean
+  :return: SSL listen port
+  """
+  ssl = None
+  ssl_listen_port = None
+  try:
+    # this can throw if SSL mbean not there
+    ssl = server.getSSL()
+    # this can throw if SSL mbean is there but enabled is false
+    ssl.getListenPort()
+    # this can throw if SSL mbean is there but enabled is false ??
+    ssl.isEnabled()
+  except:
+    pass
+
+  if ssl is not None and ssl.isEnabled():
+    if not is_server_template:
+      ssl_listen_port = ssl.getListenPort()
+    else:
+      ssl_listen_port = getRealSSLListenPort(server, ssl.getListenPort())
+  elif ssl is None and isSecureModeEnabledForDomain(domain):
+    ssl_listen_port = "7002"
+  return ssl_listen_port
+
+def get_server_template_listening_ports_from_configxml(config_xml):
+  '''
+  get_server_tempalate's listening port and ssl port from the config.xml
+  :param config_xml:         full path to config.xml
+  :return: dictionary of servertemplate ssl port and servertemplate listen port
+  '''
+  DOMTree = parse(config_xml)
+  collection = DOMTree.documentElement
+
+  templates = collection.getElementsByTagName("server-template")
+  server_template_ssls = dict()
+  server_template_ports = dict()
+
+  # if port is not specified in config.xml, set to None
+
+  for template in templates:
+    sslport = None
+    port = None
+    if template.parentNode.nodeName != 'domain':
+      continue
+    template_name = template.getElementsByTagName('name')[0].firstChild.nodeValue
+    # Get listen port
+    listen_ports = template.getElementsByTagName('listen-port')
+
+    for listen_port in listen_ports:
+      if listen_port.parentNode.nodeName == 'server-template':
+        port = listen_port.firstChild.nodeValue
+        break
+    server_template_ports[template_name] = port
+
+    # Get ssl port
+    ssls = template.getElementsByTagName('ssl')
+    if len(ssls) > 0:
+      ssl = ssls.item(0)
```

Comment:
Do we need to check if isEnabled here?

Reply:
We are okay since the code that uses the data that is collected here checks if SSL is enabled.